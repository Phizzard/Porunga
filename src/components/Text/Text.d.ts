import React from 'react'
import { Palette, Sizes, Weight, Style } from '../../types'
declare type TextProps = {
  /** Content inside Text component */
  children: React.ReactNode
  /** Class attribute for text */
  className?: string
  /** Color of text */
  color?: Palette
  /** Font size of text */
  size?: Sizes
  /** Font family of text */
  family?: string
  /** Font weight of text */
  weight?: Weight
  /** Font style of text */
  style?: Style
}
/**
 * Text
 */
declare const Text: ({
  children,
  className,
  color,
  size,
  family,
  weight,
  style,
}: TextProps) => JSX.Element
export default Text
