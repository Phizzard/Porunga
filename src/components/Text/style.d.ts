/// <reference types="react" />
/// <reference types="@emotion/core" />
declare type ContainerProps = {
  colorProp?: string
  size?: string
  family?: string
  weight?: string
  styleProp?: string
}
export declare const Container: import('@emotion/styled-base').StyledComponent<
  import('react').DetailedHTMLProps<
    import('react').HTMLAttributes<HTMLSpanElement>,
    HTMLSpanElement
  >,
  ContainerProps,
  object
>
export {}
