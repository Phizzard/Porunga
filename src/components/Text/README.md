```jsx
<Text color="brand" size="xxsmall" family="serif" weight="normal" style="normal">
  Text 1
</Text>

<Text color="light-2" size="xsmall" family="sans-serif" weight="bold" style="italic">
  Text 2
</Text>

<Text color="light-4" size="small" family="monospace" weight="bolder" style="oblique">
  Text 3
</Text>

<Text color="light-6" size="medium" family="cursive" weight="lighter" style="normal">
  Text 4
</Text>

<Text>
  Text 5
</Text>

<Text color="dark-2" size="large" family="fantasy" weight="100" style="italic">
  Text 6
</Text>

<Text color="dark-4" size="xlarge" family="Helvetica" weight="800" style="oblique">
  Text 7
</Text>

<Text color="dark-6" size="xxlarge" family="Helvetica, Arial, sans-serif" weight="unset" style="unset">
  Text 8
</Text>
```
