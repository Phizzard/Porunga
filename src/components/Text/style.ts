import styled from '@emotion/styled'

type ContainerProps = {
  colorProp?: string
  size?: string
  family?: string
  weight?: string
  styleProp?: string
}

export const Container = styled.p<ContainerProps>`
  color: ${p => p.colorProp};
  font-size: ${p => p.size};
  font-family: ${p => p.family};
  font-weight: ${p => p.weight};
  font-style: ${p => p.styleProp};
`
