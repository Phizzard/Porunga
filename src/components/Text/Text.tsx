import React from 'react'
import { useTheme } from 'emotion-theming'

import { Container } from './style'
import { Palette, Sizes, Weight, Style, ThemeConfig } from '../../types'
import { get } from '../../utils/get'

type TextProps = {
  /** Content inside Text component */
  children: React.ReactNode
  /** Class attribute for text */
  className?: string
  /** Color of text */
  color?: Palette
  /** Font size of text */
  size?: Sizes
  /** Font family of text */
  family?: string
  /** Font weight of text */
  weight?: Weight
  /** Font style of text */
  style?: Style
}

/**
 * Text
 */
const Text = ({
  children,
  className,
  color = 'dark-1',
  size = 'unset',
  family = 'sans-serif',
  weight = 'unset',
  style = 'unset',
}: TextProps) => {
  const theme: ThemeConfig = useTheme()

  const textProps = {
    className,
    colorProp: get(theme, 'root', 'colors', color),
    size: get(theme, 'text', 'sizes', size),
    family,
    weight: get(theme, 'text', 'weights', weight),
    styleProp: get(theme, 'text', 'styles', style),
  }

  return <Container {...textProps}>{children}</Container>
}

export default Text
