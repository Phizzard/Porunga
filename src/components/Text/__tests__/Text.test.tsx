import React from 'react'
import Text from '../Text'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const component = renderer.create(<Text>text</Text>)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
