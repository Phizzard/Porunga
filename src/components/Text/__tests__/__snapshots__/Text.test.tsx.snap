// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<p
  className="css-5zokvj"
>
  text
</p>
`;
